sound/soc/codecs/rt711-sdca-sdw.c

```diff
@@ -382,6 +394,35 @@ static int __maybe_unused rt711_sdca_dev_suspend(struct device *dev)
 	return 0;
 }
 
+static int __maybe_unused rt711_sdca_dev_system_suspend(struct device *dev)
+{
+	struct rt711_sdca_priv *rt711_sdca = dev_get_drvdata(dev);
+	struct sdw_slave *slave = dev_to_sdw_dev(dev);
+	int ret1, ret2;
+
+	if (!rt711_sdca->hw_init)
+		return 0;
+
+	/*
+	 * prevent new interrupts from being handled after the
+	 * deferred work completes and before the parent disables
+	 * interrupts on the link
+	 */
+	mutex_lock(&rt711_sdca->disable_irq_lock);
+	ret1 = sdw_update_no_pm(slave, SDW_SCP_SDCA_INT1,
+				SDW_SCP_SDCA_INTMASK_SDCA_0, 0);
+	ret2 = sdw_update_no_pm(slave, SDW_SCP_SDCA_INT2,
+				SDW_SCP_SDCA_INTMASK_SDCA_8, 0);
```

Comment:
looks like `->disable_irq = true` is missing

Reply:
> looks like `->disable_irq = true` is missing

And `->disable_irq = false` is missing in ` rt711_sdca_io_init`, too

Reply:
fixed, thanks for noticing this.

Reply:
@plbossart `SDW_SCP_SDCA_INT1` and `SDW_SCP_SDCA_INT2` are missed. They should be `SDW_SCP_SDCA_INTMASK1` and `SDW_SCP_SDCA_INTMASK2`

Reply:
correct, I made the same mistake twice. Gah.
Fixed now, thanks for the feedback @bardliao